app/src/main/java/com/github/geohunt/app/ui/components/challenge/BelowImageButton.kt

```diff
@@ -0,0 +1,133 @@
+package com.github.geohunt.app.ui.components.challenge
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Cancel
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.geohunt.app.R
+import com.github.geohunt.app.i18n.toSuffixedString
+import com.github.geohunt.app.ui.components.LabelledIcon
+
+@Composable
+internal fun BelowImageButton(
+    viewModel: ChallengeViewModel,
```

Comment:
A `Composable` that is not top-level should **not** have access to the `viewModel`.

Only pass down the `joinHunt` and `leaveHunt` methods instead.

Reply:
Nice catch !!